src/app/pipes/course-date-sort/course-date-sort.pipe.ts

```diff
@@ -0,0 +1,31 @@
+import { Pipe, PipeTransform } from '@angular/core';
+import * as moment from 'moment';
+
+import { ICourse } from '../../shared/models/course.model';
+
+@Pipe({
+  name: 'courseDateSort'
+})
+export class CourseDateSortPipe implements PipeTransform {
+
+  transform(courses: ICourse[], order: 'new' | 'old' = 'new'): ICourse[] {
+    const arrCopy = [...courses];
```

Comment:
Do we need a copy of array to work with it?
If so, maybe we can rename it to transformedCourses?

Reply:
At first, I was think about immutability. Sort method modify existing array, it not good in concept of data immutability. Probably we can remove copy of array here, because following of immutability is not necessary here.